js/showAllSongs.js

```diff
@@ -20,40 +21,91 @@ var comparator = function(property) {
   };
 };
 
-window.onload = function() {
+const sortSongsByArtist = function(songs) {
   const allSongs = {};
+  songs.map(function(song) {
+    // Sorting by artist
+    const artist = song["artist"];
+
+    if (allSongs.hasOwnProperty(artist)) {
+      allSongs[artist].push(song);
+    } else {
+      allSongs[artist] = [song];
+    }
+  });
+
+  let allSongsByArtist = [];
+  for (let artist in allSongs) {
+    // Song titles in alphabetical order
+    const songsByArtist = allSongs[artist].sort(comparator("title"));
+    allSongsByArtist.push({ "artist": artist, "songs": songsByArtist });
```

Comment:
I guess this isn’t new code, but `allSongs[artist].sort()` doesn’t return a separate sorted copy, it actually mutates `allSongs[artist]` and returns a reference to the same array. So this is a little misleading.

Reply:
Renamed variables so they are less confusing. Shallow copy object.